tests/language-feature/unsized-array.slang

```diff
@@ -0,0 +1,36 @@
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-compute -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-vk -compute -output-using-type
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<int> outputBuffer;
+
+int test(int arr[])
```

Comment:
This is a question.
I found it interesting that Slang would specialize a function like this and make the array sized.
It is as if the function is written like following,
```
__generic<let N:int>
int test(int arr[N])
```
How does HLSL and GLSL handle the unsized array as a function parameter natively?

Reply:
I don't know GLSL, but dxc just inlines it out (even with -O0), and doesn't support this when generating spirv.